Islandora-CLAW/controlled_access_terms
# Correct EDTF data type in JSON-LD #998

**GitHub Issue**: /Islandora-CLAW/CLAW/issues/998

See also the thread on /Islandora-CLAW/CLAW/issues/916

# What does this Pull Request do?

Enables us to dynamically set the JSON-LD date value type based on the field's value. E.g. the EDTF string "2019?" would become "2019" with the data type gYear while the string value "~2019-01" would become "2019-01" with the data type gYearMonth, etc.

Note, this *adds* the value with the appropriate date type along-side the existing string value. This allows us to store the actual EDTF value in our triple-store or repository (as strings) AND have a date type we can run queries against. 

# What's new?

* Changes `controlled_access_terms_jsonld_alter_normalized_array` to such that fields of the type `edtf` get a date value, correctly typed, in addition to the string value. 
* Changes the `controlled_access_terms_default_configurations` that use the text-based EDTF to the new field with corresponding widget and formatter.

# How should this be tested?

- Create any Person with birth dates using an EDTF value and view their JSON-LD. You will see a string value.
- Apply the PR
- Run the module update (`drush updb -y`)
- Check the person's JSON-LD again. You will see both the string value and the correctly formatted and typed date version.

# Additional Notes:

This leaves in place the existing text_edtf widget and formatters; however, any field that uses the EDTF widget will be changed to the new EDTF Field Type. Once this is merged we can do separate PRs to update islandora_demo.

# Interested parties
@Islandora-CLAW/committers (esp @DiegoPino )
